pkg/distributor/query.go

```diff
@@ -204,6 +203,10 @@ func (d *Distributor) queryIngesterStream(ctx context.Context, replicationSet ri
 
 			result.Chunkseries = append(result.Chunkseries, resp.Chunkseries...)
 			result.Timeseries = append(result.Timeseries, resp.Timeseries...)
+
+			if len(result.Chunkseries) > maxSeries || len(result.Timeseries) > maxSeries {
+				return nil, fmt.Errorf("exceeded maximum number of series in a query (limit %d)", maxSeries)
```

Comment:
What are the implications on the returned status code? I've the feeling this may be detected as a storage error and we return a 5xx error (while it should be a 4xx) but I haven't deeply checked it.

Reply:
Do you think it would work better as an httpgrpc error?

Reply:
@bboreham I don't remember how the error code propagation works in detail but I would start testing it. Do you have time/interest to work on it, otherwise we can takeover, cause we're interested into this limit too.

Reply:
I checked it and we have to return a `validation.LimitError`.